components/dynamic/DataList/index.js

```diff
@@ -49,29 +55,54 @@ const DataList = ({
     section,
   });
 
-  if (isLoading || isError) return null;
+  function renderData() {
+    const { offset, total } = data;
+    const numberOfPages = Math.ceil(total / limit) || 1;
 
-  const { offset, total } = data;
-  const numberOfPages = Math.ceil(total / limit) || 1;
+    // if our page is out of bounds, we must instead go to the last page with data
+    if (offset > total) {
+      router.push({
+        pathname,
+        query: { ...pathParams, page: numberOfPages },
+      });
+    }
 
-  // if our page is out of bounds, we must instead go to the last page with data
-  if (offset > total) {
-    router.push({
-      pathname,
-      query: { ...pathParams, page: numberOfPages },
-    });
+    if (total === 0 && isRelatedList) return null;
+    if (total === 0) return <div>{t(`search-no-results`)}</div>;
+    return children(data);
   }
 
-  if (total === 0 && isRelatedList) return null;
+  function renderPagination() {
+    const { offset, page, total } = data;
 
-  if (total === 0)
+    if (limit >= total || section === "glossaryTerms" || isRelatedList)
+      return null;
     return (
-      <Container>
-        <div>{t(`search-no-results`)}</div>
-      </Container>
+      <Pagination limit={limit} offset={offset} page={page} total={total} />
     );
+  }
 
-  return children(data);
+  if (isError) return null;
+
+  return (
+    <>
+      <Container width={width}>
+        {header && <Styled.Header>{header}</Styled.Header>}
+        {isLoading && <Loader speed="fast" isVisible />}
+        {!isLoading && data && renderData()}
+        {footerButton && (
+          <Styled.Footer>
+            <Buttonish
+              isBlock={true}
+              text={footerButton.text}
+              url={`/${footerButton.uri}`}
+            />
+          </Styled.Footer>
+        )}
```

Comment:
This footer was common to every component that used DataList so I refactored into the DataList component itself